src/generate.rs

```diff
@@ -142,52 +157,119 @@ pub fn generate_pair(opt: &GenerateOptions, rng: &mut impl Rng) -> (Sequence, Se
             let start = rng.gen_range(0..=b.len_bytes());
             b.insert(start, piece.as_str());
         }
-        ErrorModel::Repetition => {
-            let start = rng.gen_range(0..=b.len_bytes() - num_mutations);
-            let piece = b.slice(start..num_mutations).to_string();
-            let initial_length = b.len_bytes();
-            b = ropey::Rope::new();
-            // insert pieces
-            for _ in 0..initial_length / num_mutations {
-                b.insert(b.len_bytes(), piece.as_str());
-            }
-            // insert the segment of the last piece
-            b.insert(b.len_bytes(), &piece[initial_length - b.len_bytes()..]);
-        }
         ErrorModel::Insert => {
-            let start = rng.gen_range(0..num_mutations - num_mutations);
-            let piece = b.slice(start..start + num_mutations).to_string();
+            let start = rng.gen_range(0..b.len_bytes() - num_mutations);
+            let piece = b.slice(start..start + num_mutations / 2).to_string();
```

Comment:
no `/2` here, but add it for `DoubleInsert`, so that the total distance is `e*n` again